lib/scaleway/bootstrap/ks-gen-kubeconfig.tf

```diff
@@ -0,0 +1,6 @@
+resource "local_file" "kubeconfig" {
+  filename = "test-cluster/test-cluster.yaml" # TODO(benjaminch): use "{{ s3_kubeconfig_bucket }}/${var.kubernetes_cluster_id}.yaml"
+  content = scaleway_k8s_cluster.kubernetes_cluster.kubeconfig[0].config_file
+  file_permission = "0644"
+  depends_on = [scaleway_k8s_pool.kubernetes_cluster_worker] # TODO(benjaminch): use `scw_ks_worker_node in scw_ks_worker_nodes` to get all nodes names
```

Comment:
don't depends on worker nodes (never ever) as they can change and dependency is hard in TF (if something depends on it and dependency is not there, it's deleted). In addition to this, the kubeconfig depends ONLY on masters and not workers, so you can safely update this dependency to master